src/Security/DiscordAuthenticator.php

```diff
@@ -113,6 +113,8 @@ public function getUser($credentials, UserProviderInterface $userProvider): User
 
         try {
             $user = $userProvider->loadUserByUsername($externalId);
+            $user->setAvatarHash($discordResourceOwner->getAvatarHash());
+            $this->em->flush();
```

Comment:
I think we should move this `flush()` outside the try/catch block as it will always be needed no matter if user was found or the new user was created.